source/Sashimi.AzureCloudService/Endpoints/AzureCloudServiceDeploymentTargetTypeProvider.cs

```diff
@@ -24,5 +25,12 @@ public IActionHandler HealthCheckActionHandlerForTargetType()
         {
             return new AzureCloudServiceHealthCheckActionHandler();
         }
+
+        public IEnumerable<(string key, object value)> GetMetric(IReadOnlyCollection<Endpoint> endpoints)
+        {
+            var total = endpoints.Count(m => m is AzureCloudServiceEndpoint);
+
+            yield return ("azurecloudservices", total);
```

Comment:
These are a given. I wonder if we could have something like `string TotalCountMetricName {get;}` because the caller should already know from the `DeploymentTargetType` property of this instance what type is should be adding up, it just isn't sure what name to give it.
Then maybe `GetMetric` becomes `GetAdditionalMetrics`, or something like that? For the case like ServiceFabric below. Passing all endpoints in here doesn't feel right to me either, this should be filtered by the caller. `GetMetric(IReadOnlyCollection<T> endpoints)` would feel more natural, but trickier to call, knowing how the calls would have to be made. That being said, some less pretty code inside server to make the consumer's life clearer doesn't feel like a bad thing.

Reply:
The same applies to accounts, maybe we should have 
```
string TotalCountMetricName {get;}
IEnumerable<(string key, object value)> GetAdditionalMetrics(IReadOnlyCollection<AccountDetails> details)
```

And you're right Shannon, it's easy to filter on the server side

Reply:
As I said, I just followed what is in accounts, so whatever we decide on we need to do the same for accounts.

Reply:
Fixed